mlos_bench/mlos_bench/optimizer/base_optimizer.py

```diff
@@ -98,17 +98,36 @@ def __init__(self, tunables: TunableGroups, config: dict):
                 raise ValueError("Cannot specify both 'maximize' and 'minimize'.")
             self._opt_sign = -1
 
+    def __repr__(self) -> str:
+        opt_direction = 'min' if self._opt_sign > 0 else 'max'
+        return f"{self.__class__.__name__}:{opt_direction}({self._opt_target})"
+
+    @property
+    def target(self) -> str:
+        """
+        The name of the target metric to optimize.
+        """
+        return self._opt_target
+
     @abstractmethod
-    def bulk_register(self, data: List[dict]):
+    def bulk_register(self, configs: List[dict], scores: List[float]) -> bool:
         """
         Pre-load the optimizer with the bulk data from previous experiments.
 
         Parameters
         ----------
-        data : List[dict]
-            Records of tunable values and benchmark scores from other experiments.
-            The data is expected to be in `pandas.DataFrame.to_dict('records')` format.
+        configs : List[dict]
+            Records of tunable values from other experiments.
+        scores : List[float]
```

Comment:
In the previous `data` did we also support a `status` column in addition to scores?
For instance, I am wondering how we handle trials that failed and don't have a score, especially since the typical strategy for that is to compute a "bad" score based on all *real* data we've seen so far.
Not storing an explicit score leaves us some flexibility to manufacture the score in the future which is useful for merge experiments where we are trying to reuse data.

Reply:
You're right about the scores - none of our optimizers currently supports this, but we should retrieve the failed trials data anyway. I'll fix that

Reply:
Mostly makes sense.  Not sure if Status should be Optional in this case though.
On the `mlos_core` side it won't know anything about it which is fine.
It's on this shim side where we can compute fabricated "bad" scores based on real data for the "FAILED" status ones.